common/src/main/java/io/netty5/util/internal/PlatformDependent0.java

```diff
@@ -224,14 +227,39 @@
                 final Object maybeDirectBufferConstructorHandle =
                         AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
                             try {
-                                final Constructor<?> constructor =
-                                        direct.getClass().getDeclaredConstructor(long.class, int.class, Object.class);
-                                Throwable cause = ReflectionUtil.trySetAccessible(constructor, true);
-                                if (cause != null) {
-                                    return cause;
-                                }
-                                return MethodHandles.lookup().unreflectConstructor(constructor);
-                            } catch (NoSuchMethodException | SecurityException | IllegalAccessException e) {
+                                return Arrays
+                                        .stream(direct.getClass().getDeclaredConstructors())
+                                        .filter(declaredConstructor -> {
+                                            int parameterCount = declaredConstructor.getParameterCount();
+                                            if (parameterCount != 3 && parameterCount != 4) {
+                                                return false;
+                                            }
+                                            Parameter[] parameters = declaredConstructor.getParameters();
+                                            return parameters[0].getType() == long.class
+                                                    && parameters[1].getType() == int.class
+                                                    && parameters[2].getType() == Object.class;
+                                        })
+                                        .findFirst()
+                                        .map(constructor -> {
+                                            try {
+                                                Throwable cause = ReflectionUtil
+                                                        .trySetAccessible(constructor, true);
+                                                if (cause != null) {
+                                                    return cause;
+                                                }
+                                                MethodHandle constructorHandle = MethodHandles.lookup()
+                                                        .unreflectConstructor(constructor);
+                                                if (constructor.getParameterCount() == 4) {
+                                                    Object[] nullArgs = new Object[] { null };
+                                                    MethodHandles.insertArguments(constructorHandle, 3, nullArgs);
```

Comment:
`insertArguments` does not modify the given method handle, but instead returns a new method handle with the added behaviour.
We need to assign the result of `insertArguments` to the `constructorHandle` variable.

Reply:
Done